native/level1double.go

```diff
@@ -127,6 +127,7 @@ func (Implementation) Idamax(n int, x []float64, incX int) int {
 	idx := 0
 	max := math.Abs(x[0])
 	if incX == 1 {
+		x = x[:n]
```

Comment:
Good point. Moved into range statement. 

By explicit for loop I assume you mean the non-range one below? It needs to be an explicit for loop because we only examine the strided elements. It's not the maximum in the slice, it's the maximum of the strided elements in the slice.